resources/qml/MessageView.qml

```diff
@@ -192,6 +192,9 @@ Item {
         Shortcut {
             sequence: StandardKey.Cancel
             onActivated: {
+                if(room.input.uploads)
+                room.input.declineUploads();
+                
                 if (chat.model.reply)
```

Comment:
```suggestion
                    room.input.declineUploads();
                else if (chat.model.reply)
```

Otherwise you cancel multiple things at the same time :3 

Apart from that this looks good to me!

Reply:
I think I also need to replace the first if-condition to check for the uploads length since room.input.upload is defined even when their is no file queued on the upload list

Reply:
Right, that makes sense,we do a length check elsewhere too :3 

Just add .length > 0 then